[improvement] Dependencies update

veronikaslc opened the pull request:
The focus of this PR is [migration from 17 to 18 version of React.](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html)
No breaking changes in the UI or functionality expected except faster rendering.
To test please inspect overall app.

veronikaslc committed:
[improvement] Depenencies update

marta- commented:
Putting any testing on hold until we merge more of the pending UI changes. Then we will rebase and can test the dependency bump on a version that is closer to what will be released.

marta- commented:
@veronikaslc Please rebase on the latest dev for testing

sashaandjic commented:
WSOD happens when trying to access the following in the Admin:
- Clinic Dashboard
- Patient Access
- Patient Portal Survey Instructions

veronikaslc committed:
[improvement] Dependencies update
useEffect must not return anything besides a function, otherwise we have the huge crash in destroy phase

sashaandjic reviewed (approved):
tested the latest changes

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/homepage/src/main/frontend/src/adminDashboard/AdminConfigScreen.jsx:
```diff
@@ -108,8 +108,8 @@ function AdminConfigScreen(props) {
-  useEffect(() => getConfig(), []);
-  useEffect(() => hasChanges && setConfigIsInitial(false), [hasChanges]);
+  useEffect(() => {getConfig()}, []);
+  useEffect(() => {hasChanges && setConfigIsInitial(false)}, [hasChanges]);
```
Why this change? Is it because the missing `{}` cause an issue? There is similar bracketless code in `Note.jsx`, `DicomQuestion.jsx` and `ClinicDashboard.jsx`, but I don't see any errors when using that code without changes.

veronikaslc replied:
Without `{}` there is a warning 
`react_devtools_backend.js:2655 Warning: useEffect must not return anything besides a function, which is used for clean-up. You returned: false at AdminConfigScreen `

veronikaslc replied:
in Note.jsx, DicomQuestion.jsx and ClinicDashboard.jsx probably there is no warning because `useEffect()` did not get a chance to be triggered. However sintactically `{}` are recommended for a clean code.

sdumitriu reviewed (approved):
I see more warnings about unmet dependencies when building the frontend, mostly from MUI and its related modules depending on React 16|17. I guess MUI didn't yet switch to React 18.

veronikaslc commented:
> I see more warnings about unmet dependencies when building the frontend, mostly from MUI and its related modules depending on React 16|17. I guess MUI didn't yet switch to React 18.

Afaik Material UI v5 is the only version that fully supports React 18 nad we are on `5.11`, should be good.  Somehow my build does not have any dependency conflicts. Did you delete `yarn.lock` file before build?

sdumitriu committed:
[misc] Fix React error after saving and viewing a DICOM form

sdumitriu commented:
> > I see more warnings about unmet dependencies when building the frontend, mostly from MUI and its related modules depending on React 16|17. I guess MUI didn't yet switch to React 18.
> 
> Afaik Material UI v5 is the only version that [supports React 18](https://github.com/mui/material-ui/milestone/45) nad we are on `5.11`, should be good. Somehow my build does not have any dependency conflicts. Did you delete `yarn.lock` file before build?

This is what I get:

```
[INFO] [3/4] Linking dependencies...
[INFO] warning " > @mui/styles@5.11.2" has incorrect peer dependency "react@^17.0.0".
[INFO] warning " > material-table@2.0.3" has unmet peer dependency "@date-io/core@^2.13.1".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "@mui/styles@5.5.0".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "react@^17.0.2".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "react-dom@^17.0.2".
[INFO] warning "material-table > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react@^16.8.5 || ^17.0.0".
[INFO] warning "material-table > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react-dom@^16.8.5 || ^17.0.0".
[INFO] warning "react-router-dom > react-router > mini-create-react-context@0.4.1" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning " > formik-material-fields@0.0.6" has incorrect peer dependency "react@^16.3.0".
[INFO] warning " > formik-material-fields@0.0.6" has incorrect peer dependency "react-dom@^16.3.0".
[INFO] warning " > @mui/x-date-pickers@5.0.13" has unmet peer dependency "@mui/system@^5.4.1".
[INFO] warning " > @material-ui/core@4.11.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
[INFO] warning " > @material-ui/core@4.11.2" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/utils@4.11.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/utils@4.11.3" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/system@4.12.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/system@4.12.2" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/styles@4.11.5" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
[INFO] warning "@material-ui/core > @material-ui/styles@4.11.5" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
[INFO] warning " > @material-ui/icons@4.11.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
[INFO] warning " > @material-ui/icons@4.11.2" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
[INFO] warning " > react-to-print@2.14.4" has incorrect peer dependency "react@^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning " > react-to-print@2.14.4" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning "@uiw/react-md-editor > @uiw/react-markdown-preview > react-markdown@8.0.5" has unmet peer dependency "@types/react@>=16".
[INFO] warning " > html-webpack-plugin@5.5.0" has incorrect peer dependency "webpack@^5.20.0".
```

veronikaslc commented:
Here is what I have for dependencies. Somehow much less for MUI than your build.

`[INFO] Running 'yarn ' in C:\Users\veron\Git\cards\aggregated-frontend\src\main\frontend
[INFO] yarn install v1.22.17
[INFO] [1/4] Resolving packages...
[INFO] warning Resolution field "@babel/runtime@7.20.7" is incompatible with requested version "@babel/runtime@^7.20.13"
[INFO] warning Resolution field "@babel/runtime@7.20.7" is incompatible with requested version "@babel/runtime@^7.20.13"
[INFO] warning Resolution field "@babel/runtime@7.20.7" is incompatible with requested version "@babel/runtime@^7.20.13"
[INFO] warning Resolution field "@babel/runtime@7.20.7" is incompatible with requested version "@babel/runtime@^7.20.13"
[INFO] [2/4] Fetching packages...
[INFO] [3/4] Linking dependencies...
[INFO] warning " > @mui/styles@5.11.2" has incorrect peer dependency "react@^17.0.0".
[INFO] warning " > @mui/x-date-pickers@5.0.13" has unmet peer dependency "@mui/system@^5.4.1".
[INFO] warning "react-router-dom > react-router > mini-create-react-context@0.4.1" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning " > react-to-print@2.14.4" has incorrect peer dependency "react@^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning " > react-to-print@2.14.4" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0 || ^17.0.0".
[INFO] warning " > material-table@2.0.3" has unmet peer dependency "@date-io/core@^2.13.1".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "@mui/styles@5.5.0".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "react@^17.0.2".
[INFO] warning " > material-table@2.0.3" has incorrect peer dependency "react-dom@^17.0.2".
[INFO] warning "material-table > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react@^16.8.5 || ^17.0.0".
[INFO] warning "material-table > react-beautiful-dnd@13.1.0" has incorrect peer dependency "react-dom@^16.8.5 || ^17.0.0".
[INFO] warning " > material-ui-dropzone@3.5.0" has unmet peer dependency "@material-ui/core@^4.0.0".
[INFO] warning " > material-ui-dropzone@3.5.0" has unmet peer dependency "@material-ui/icons@^4.0.0".
[INFO] warning "@uiw/react-md-editor > @uiw/react-markdown-preview > react-markdown@8.0.5" has unmet peer dependency "@types/react@>=16".
[INFO] warning " > semantic-ui-react@0.83.0" has incorrect peer dependency "react@^16.0.0".
[INFO] warning " > semantic-ui-react@0.83.0" has incorrect peer dependency "react-dom@^16.0.0".
[INFO] warning " > formik-material-fields@0.0.6" has unmet peer dependency "@material-ui/core@>=3".
[INFO] warning " > formik-material-fields@0.0.6" has incorrect peer dependency "react@^16.3.0".
[INFO] warning " > formik-material-fields@0.0.6" has incorrect peer dependency "react-dom@^16.3.0".
[INFO] warning " > html-webpack-plugin@5.5.0" has incorrect peer dependency "webpack@^5.20.0".
[INFO] [4/4] Building fresh packages...
[INFO] success Saved lockfile.
[INFO] Done in 273.26s.`

veronikaslc commented:
I'm wondering from where your build got `@material-ui/core`. This package is deprecated and belongs to Material UI v4.

veronikaslc commented:
`@mui/styles` is the only thing with real MUI unmet dependency. As they say, no guarantee of compatability however I did not noticed broken styles from what we use, and we do use alot of `makeStyles` from `@mui/styles`.

marta- commented:
Sections labels now have a white background inside a gray background, instead of all gray.

![image](https://user-images.githubusercontent.com/651980/220502837-c72f351f-0784-4ab7-a77d-f38d4d37f06e.png)

sashaandjic commented:
> Sections labels now have a white background inside a gray background. It should be all gray.

Confirm the defect. It appears in the latest rendition of this PR. Below are the screenshots of an instance running with changes prior to today vs today's PR:
![image](https://user-images.githubusercontent.com/23243244/220509848-8e3f9e31-4f93-41c9-912c-4412291ab02a.png)

marta- commented:
> > Sections labels now have a white background inside a gray background. It should be all gray.
> 
> Confirming the defect. It appears in the latest rendition of this PR.

Today it was rebased on the latest dev. The _before_ screenshots show a state that predates the survey updates (including shorter titles), and I'm inclined to believe it also doesn't contain #1316, which introduced formatting in section labels. It appears that in this update PR the background is always white for rendered markdown text.

marta- committed:
[misc] Bugfix: make the FormattedText background transparent again

marta- commented:
> Sections labels now have a white background inside a gray background. It should be all gray.

Should be fixed by 8ba5296.

marta- commented:
> No breaking changes in the UI or functionality expected except faster rendering.

@veronikaslc I'm not noticing any speedup. Is there a particular scenario I could try where the rendering is noticeably faster?

sashaandjic reviewed (approved):
tested - section labels are gray again
![image](https://user-images.githubusercontent.com/23243244/220664987-5c398ca6-1160-4aaa-8284-c26d4422f562.png)
![image](https://user-images.githubusercontent.com/23243244/220665024-873ade27-8bea-41e2-944d-f70d631e0a48.png)

marta- commented:
The text of descriptions of sections, questions, options is now black instead of gray.

marta- committed:
[misc] Bugfix: make FormattedText use the specified color

veronikaslc commented:
> > No breaking changes in the UI or functionality expected except faster rendering.
> 
> @veronikaslc I'm not noticing any speedup. Is there a particular scenario I could try where the rendering is noticeably faster?

Not I know of on top of my head.

sashaandjic reviewed (approved):
tested

sashaandjic reviewed (approved):
tested
